doc/srv6/srv6_static_config_hld.md

```diff
@@ -0,0 +1,172 @@
+# Static Configuration of SRv6 in SONiC HLD  
+
+# Table of Contents
+
+- [Revision](#revision)
+- [Definition/Abbreviation](#definitionabbreviation)
+- [About This Manual](#about-this-manual)
+- [1 Introuduction and Scope](#1-introuduction-and-scope)
+- [2 Feature Requirements](#2-feature-requirements)
+- [2.1 Functional Requirements](#21-functional-requirements)
+- [2.2 Configuration and Managment Requirements](#22-configuration-and-management-requirements)
+- [2.3 Warm Boot Requirements](#23-warm-boot-requirements)
+- [3 Feature Design](#3-feature-design)
+- [3.1 New Table in ConfigDB](#31-new-table-in-configdb)
+- [3.2 Bgpcfgd Changes](#32-bgpcfgd-changes)
+- [3.3 YANG Model](#33-yang-model)
+- [4 Unit Test](#4-unit-test)
+- [5 References ](#5-references) 
+
+# Revision
+
+| Rev  |   Date    |           Author           | Change Description      |
+| :--: | :-------: | :------------------------: | :---------------------: |
+| 0.1  | 12/5/2024 |       Changrong Wu         |  Initial version        |
+
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+
+| ****Term**** | ****Meaning**** |
+| -------- | ----------------------------------------- |
+| BGP  | Border Gateway Protocol |
+| SID  | Segment Identifier  |
+| SRv6 | Segment Routing IPv6  |
+| SDN | Software Defined Network |
+| uSID | Micro Segment |
+| VRF  | Virtual Routing and Forwarding  |
+
+# About this Manual
+
+This document provides general information about the design of the enhancements in SONiC to support static configuration of Segmentation Routing over IPv6 protocol, which is crucial for SRv6 SDN deployment (without usage of BGP).
+
+# 1 Introuduction and Scope
+
+This document describes the high-level design of the new features in SONiC to support SRv6 SDN.
+The new features include the addtion of a new table in CONFIG_DB to allow static configuration of SRv6 and the enhancement of bgpcfgd to program FRR with input from CONFIG_DB.
```

Comment:
Hi @BYGX-wcr, thanks for the HLD to support SRv6 uSID static configuration. 

The static SRv6 uSID configuration model in FRR as follows: 
-	Configure an SRv6 locator and identify its parameters (name, prefix, block-len, node-len, func-bits). The CLI for this is already in FRR mainline. 
-	Configure a static SRv6 SID under the SRv6 locator and identify its parameters (behavior, vrf, etc.). The CLI for this is being added in this FRR PR (https://github.com/FRRouting/frr/pull/16894)

In the current HLD, the configuration model is different and hence can’t be mapped to the current SRv6 uSID configuration model in FRR. 

To resolve this, I would propose the following: 
-	Create a new table (SRV6_MY_LOCATOR) for the SRv6 locator config and its parameters (name, prefix, block-len, node-len, func-bits).
-	In the SRV6_MY_SID_TABLE, you add the locator as new parameter for the SID. The locator parameter points to one of the locators in SRV6_MY_LOCATOR via the locator’s name. This way you can map the config to FRR config.

FRR inherits the (block-len, node-len, func-bits) of the locator and install the static SID into SONiC APPL_DB via fpm-sonic-dplane. This is already mainline in FRR and SONiC.

Reply:
The locator PR is added via https://github.com/sonic-net/sonic-buildimage/pull/20954 to frrcfgd. 

The sample config db json could be found at 

https://github.com/sonic-net/sonic-mgmt/blob/master/ansible/vars/configdb_jsons/7nodes_cisco/PE3.json#L435

Reply:
@ahsalam: I understand the logic being used by SRV6_MY_LOCATOR. However our current config db schema is good enough to program FRR. We do not want indirection logic using 2 different config tables. If you see in below example we have all the information to program frr.  We will parse ipv6 address and extract the opcode `prefix` to FRR.  Keeping one table in synch with APP_DB format is easier for us.

Reply:
@ahsalam / @eddieruan-alibaba : We also planning to remove `vrf` from config db schema as our current use case is to use `SAI_MY_SID_ENTRY_ATTR_TUNNEL_ID` for uDT46 usecase. Currently Tunnel Decap Schema https://github.com/sonic-net/SONiC/blob/56a689c40f3d68536a5ed06b592080003dbd8c7c/doc/decap/subnet_decap_HLD.md?plain=1#L149 does not support vrf for overlay interface so their is no point of specifying VRF as needed by `SAI_MY_SID_ENTRY_ATTR_VRF`.

Reply:
DT46 will use default VRF.  Our use case is not a typical L3VPN use case.